### PR TITLE
Populate Default Version

### PR DESCRIPTION
**Pull request by amy**

### CODE REVIEW

**Comment by amy** [comment]
Some integration tests were removed because they are no longer in [test-catalog](https://github.com/rancher/test-catalog). The original removal in test-catalog was because it conflicted with appropriate catalog testing for `v1.6` in terms of loading the catalog itself.

File: service/utils.go [hunk 1]
```diff
@@ -88,24 +88,24 @@ func catalogResource(catalog model.Catalog, apiContext *api.ApiContext, envId st
...
 
-func templateDefaultVersion(template model.Template, catalogName string, apiContext *api.ApiContext) (string, string) {
+func templateDefaultVersion(template model.Template, catalogName string, apiContext *api.ApiContext) (string, string, string) {
 	var defaultTemplateVersionId string
 	for _, version := range template.Versions {
 		if version.Version == template.DefaultVersion {
 			defaultTemplateVersionId = generateVersionId(catalogName, template, version)
 			selfLink := apiContext.UrlBuilder.ReferenceByIdLink("templates", defaultTemplateVersionId)
-			return selfLink, defaultTemplateVersionId
+			return selfLink, defaultTemplateVersionId, version.Version
```
Comment (joshwget) [inline service/utils.go]: This case is where `DefaultVersion` comes from `config.yml`, correct?

File: service/utils.go [hunk 2]
```diff
@@ -88,24 +88,24 @@ func catalogResource(catalog model.Catalog, apiContext *api.ApiContext, envId st
...
 		}
 	}
 
 	sort.Sort(model.Versions(template.Versions))
 	if len(template.Versions) != 0 {
 		defaultTemplateVersionId = generateVersionId(catalogName, template, template.Versions[len(template.Versions)-1])
 		selfLink := apiContext.UrlBuilder.ReferenceByIdLink("templates", defaultTemplateVersionId)
-		return selfLink, defaultTemplateVersionId
+		return selfLink, defaultTemplateVersionId, template.Versions[len(template.Versions)-1].Version
```
Comment (joshwget) [inline service/utils.go]: And this is where it is detected automatically? This takes min/max Rancher version into account, right?

File: service/utils.go (hunk 1 above)
Comment (amy) [reply service/utils.go]: Yup

File: service/utils.go (hunk 2 above)
Comment (amy) [reply service/utils.go]: Tested it out, it does take min/max Rancher version into account.

**Comment by joshwget** [comment]
LGTM

**Pull request merged**